src/components/themed/TransactionListTop.tsx

```diff
@@ -156,10 +156,10 @@ export class TransactionListTopComponent extends React.PureComponent<Props, Stat
   handleOpenWalletListModal = () => {
     triggerHaptic('impactLight')
     const { navigation } = this.props
-    Airship.show<WalletListResult>(bridge => <WalletListModal bridge={bridge} headerTitle={s.strings.select_wallet} />).then(
+    Airship.show<WalletListResult>(bridge => <WalletListModal bridge={bridge} headerTitle={s.strings.select_wallet} navigation={navigation} />).then(
       ({ walletId, currencyCode }: WalletListResult) => {
         if (walletId != null && currencyCode != null) {
-          navigation.setParams({ currencyCode, walletId })
+          navigation.replace('transactionList', { currencyCode, walletId })
```

Comment:
Did you test that `.replace` behaves the same as `.setParams`? I wonder if this is equivelent. It'd be safer to keep the navigation prop type as it is so you can just use `setParams` for the scene that TransactionListTop expects to be mounted on.

Reply:
Still needs fixing.